src/parquet/column/levels.h

```diff
@@ -82,5 +83,32 @@ class PARQUET_EXPORT LevelDecoder {
   std::unique_ptr<BitReader> bit_packed_decoder_;
 };
 
+// ----------------------------------------------------------------------
+// Definition and Repetition level vectors
```

Comment:
Actually it is being used (except `StructRepLevels` which is there for future repeated+struct impl, maybe I should remove it).
I have avoided converting all of the existing code that uses `uint16[]` to use these datatypes though because it has occured to me that these things may change with the full complex schema impl. That's is also why I've only put there some basic encapsulations of the logic in the form of static factory functions.